src/plugins/orangehrmCorporateDirectoryPlugin/Dao/EmployeeDirectoryDao.php

```diff
@@ -0,0 +1,121 @@
+ * the GNU General Public License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * OrangeHRM is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace OrangeHRM\CorporateDirectory\Dao;
+
+use OrangeHRM\Core\Dao\BaseDao;
+use OrangeHRM\Core\Traits\Service\TextHelperTrait;
+use OrangeHRM\CorporateDirectory\Dto\EmployeeDirectorySearchFilterParams;
+use OrangeHRM\Entity\Employee;
+use OrangeHRM\ORM\QueryBuilderWrapper;
+
+class EmployeeDirectoryDao extends BaseDao
+{
+    use TextHelperTrait;
+
+    /**
+     * @param EmployeeDirectorySearchFilterParams $employeeDirectorySearchParamHolder
+     * @return Employee[]
+     */
+    public function getEmployeeList(EmployeeDirectorySearchFilterParams $employeeDirectorySearchParamHolder): array
+    {
+        $qb = $this->getEmployeeListQueryBuilderWrapper($employeeDirectorySearchParamHolder)->getQueryBuilder();
+        return $qb->getQuery()->execute();
+    }
+
+    /**
+     * @param EmployeeDirectorySearchFilterParams $employeeDirectorySearchParamHolder
+     * @return QueryBuilderWrapper
+     */
+    protected function getEmployeeListQueryBuilderWrapper(
+        EmployeeDirectorySearchFilterParams $employeeDirectorySearchParamHolder
+    ): QueryBuilderWrapper {
+        $q = $this->createQueryBuilder(Employee::class, 'employee');
+        $q->leftJoin('employee.jobTitle', 'jobTitle');
+        $q->leftJoin('employee.subDivision', 'subunit');
+        $q->leftJoin('employee.locations', 'location');
+
+        $this->setSortingAndPaginationParams($q, $employeeDirectorySearchParamHolder);
+
+        if (is_null($employeeDirectorySearchParamHolder->getIncludeEmployees()) ||
+            $employeeDirectorySearchParamHolder->getIncludeEmployees() ===
+            EmployeeDirectorySearchFilterParams::INCLUDE_EMPLOYEES_ONLY_CURRENT
+        ) {
+            $q->andWhere($q->expr()->isNull('employee.employeeTerminationRecord'));
+        } elseif (
+            $employeeDirectorySearchParamHolder->getIncludeEmployees() ===
+            EmployeeDirectorySearchFilterParams::INCLUDE_EMPLOYEES_ONLY_PAST
+        ) {
+            $q->andWhere($q->expr()->isNotNull('employee.employeeTerminationRecord'));
+        }
+
+        if (!is_null($employeeDirectorySearchParamHolder->getEmpNumber())) {
+            $q->andWhere('employee.empNumber = :empNumber')
+                ->setParameter('empNumber', $employeeDirectorySearchParamHolder->getEmpNumber());
```

Comment:
this is wrong while `$employeeDirectorySearchParamHolder->getEmpNumber()` is an array

Reply:
Done